In [2]:
!pip install ydata-profiling
import pandas as pd
import numpy as np
import seaborn as sns

### Step 1 — Load & preview

In [2]:
df = sns.load_dataset('titanic')
print(df.head())
print(df.shape)

   survived  pclass     sex   age  sibsp  parch     fare embarked  class  \
0         0       3    male  22.0      1      0   7.2500        S  Third   
1         1       1  female  38.0      1      0  71.2833        C  First   
2         1       3  female  26.0      0      0   7.9250        S  Third   
3         1       1  female  35.0      1      0  53.1000        S  First   
4         0       3    male  35.0      0      0   8.0500        S  Third   

     who  adult_male deck  embark_town alive  alone  
0    man        True  NaN  Southampton    no  False  
1  woman       False    C    Cherbourg   yes  False  
2  woman       False  NaN  Southampton   yes   True  
3  woman       False    C  Southampton   yes  False  
4    man        True  NaN  Southampton    no   True  
(891, 15)


### Step 2 — Shape & types

In [3]:
print('Shape:', df.shape)
print(df.info())
print('Memory (KB):', df.memory_usage(deep=True).sum() / 1024)

Shape: (891, 15)
<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 15 columns):
 #   Column       Non-Null Count  Dtype   
---  ------       --------------  -----   
 0   survived     891 non-null    int64   
 1   pclass       891 non-null    int64   
 2   sex          891 non-null    object  
 3   age          714 non-null    float64 
 4   sibsp        891 non-null    int64   
 5   parch        891 non-null    int64   
 6   fare         891 non-null    float64 
 7   embarked     889 non-null    object  
 8   class        891 non-null    category
 9   who          891 non-null    object  
 10  adult_male   891 non-null    bool    
 11  deck         203 non-null    category
 12  embark_town  889 non-null    object  
 13  alive        891 non-null    object  
 14  alone        891 non-null    bool    
dtypes: bool(2), category(2), float64(2), int64(4), object(5)
memory usage: 80.7+ KB
None
Memory (KB): 313.72265625


### Data Audit

**Numeric Columns (int64, float64):**
- survived  
- pclass  
- age  
- sibsp  
- parch  
- fare  

**Categorical Columns (object, category, bool):**
- sex  
- embarked  
- class  
- who  
- adult_male  
- deck  
- embark_town  
- alive  
- alone  

**Columns with Null Values:**
- age (714 non-null out of 891)  
- embarked (889 non-null out of 891)  
- deck (203 non-null out of 891)  
- embark_town (889 non-null out of 891)  


###  Step-3 Missing values & duplicates

In [4]:
missing = df.isnull().sum().sort_values(ascending=False)
missing_pct = (missing / len(df) * 100).round(2)
print(pd.DataFrame({'Missing Count': missing, 'Missing %': missing_pct}))
print('Duplicate rows:', df.duplicated().sum())

             Missing Count  Missing %
deck                   688      77.22
age                    177      19.87
embarked                 2       0.22
embark_town              2       0.22
sex                      0       0.00
pclass                   0       0.00
survived                 0       0.00
fare                     0       0.00
parch                    0       0.00
sibsp                    0       0.00
class                    0       0.00
adult_male               0       0.00
who                      0       0.00
alive                    0       0.00
alone                    0       0.00
Duplicate rows: 107


### Step 4 — Descriptive statistics
Stats + skewness + kurtosis

In [5]:
print(df.describe().T)
numeric_cols = df.select_dtypes(include='number').columns
print('Skewness:')
print(df[numeric_cols].skew().round(3))
print('Kurtosis:')
print(df[numeric_cols].kurtosis().round(3))

          count       mean        std   min      25%      50%   75%       max
survived  891.0   0.383838   0.486592  0.00   0.0000   0.0000   1.0    1.0000
pclass    891.0   2.308642   0.836071  1.00   2.0000   3.0000   3.0    3.0000
age       714.0  29.699118  14.526497  0.42  20.1250  28.0000  38.0   80.0000
sibsp     891.0   0.523008   1.102743  0.00   0.0000   0.0000   1.0    8.0000
parch     891.0   0.381594   0.806057  0.00   0.0000   0.0000   0.0    6.0000
fare      891.0  32.204208  49.693429  0.00   7.9104  14.4542  31.0  512.3292
Skewness:
survived    0.479
pclass     -0.631
age         0.389
sibsp       3.695
parch       2.749
fare        4.787
dtype: float64
Kurtosis:
survived    -1.775
pclass      -1.280
age          0.178
sibsp       17.880
parch        9.778
fare        33.398
dtype: float64


### Step 5 - Categorical summary
Value counts per category

In [6]:
for col in ['survived', 'sex', 'pclass', 'embarked']:
    print(f'\n--- {col} ---')
    print(df[col].value_counts())
    print(f'Unique values: {df[col].nunique()}')


--- survived ---
survived
0    549
1    342
Name: count, dtype: int64
Unique values: 2

--- sex ---
sex
male      577
female    314
Name: count, dtype: int64
Unique values: 2

--- pclass ---
pclass
3    491
1    216
2    184
Name: count, dtype: int64
Unique values: 3

--- embarked ---
embarked
S    644
C    168
Q     77
Name: count, dtype: int64
Unique values: 3


### Step 6 — Automated HTML report
Generate ydata-profiling report

In [14]:
!pip install setuptools
!pip install -U ydata-profiling


  Using cached ydata_profiling-4.18.4-py2.py3-none-any.whl.metadata (23 kB)
Using cached ydata_profiling-4.18.4-py2.py3-none-any.whl (400 kB)
  Attempting uninstall: ydata-profiling
    Found existing installation: ydata-profiling 0.0.dev0
    Uninstalling ydata-profiling-0.0.dev0:


ERROR: Could not install packages due to an OSError: [WinError 5] Access is denied: 'd:\\anaconda\\envs\\growurk-ds\\scripts\\ydata_profiling.exe'
Consider using the `--user` option or check the permissions.

